app/models/concerns/teacher.rb

```diff
@@ -139,4 +141,54 @@ def update_teacher params
     end
     response
   end
-end
\ No newline at end of file
+
+  def is_premium?
+    subscriptions
+      .where("subscriptions.expiration >= ?", Date.today)
+      .where("subscriptions.account_limit >= ?", self.my_students.length)
+      .any?
+  end
+
+  def my_students
+    User.joins("JOIN classrooms ON users.classcode = classrooms.code")
+        .joins("JOIN users AS teachers ON teachers.id = classrooms.teacher_id")
+        .where("teachers.id = ?", self.id)
+  end
+
+  def is_trial_expired?
+    acss = self.teachers_activity_sessions_since_date(TRIAL_START_DATE)
+    acss.count > TRIAL_LIMIT
+  end
+
+  def teachers_activity_sessions_since_date date
+    ActivitySession.where(user: self.my_students)
+                   .where("completed_at >= ?", date)
+  end
+
+  def trial_activities_display_ratio
+    "#{teachers_activity_sessions_since_date(trial_start_date).count} / #{trial_limit}"
+  end
+
+  def trial_activities_numerical_ratio
```

Comment:
@kriskelly not sure which of the above methods you are referring to. I agree with you about the method #trial_activities_numerical_ratio, and will move that to the helper. The #is_trial_expired? is used in a method #premium_state in the Teacher module which itself is used in the controller, so I think it makes sense to keep #is_trial_expired? (as well as its helper #teachers_activity_sessions_since_date) in the Teacher module.